### PR TITLE
Reorganization of extension.js code - OOP approach

### PR DESCRIPTION
**Pull request by kowsen**
After looking at the code in extension.js, I felt like it would be easier to maintain and build upon if the code were more modular.

I split it into a few different objects, using a sort of MVC philosophy. I treated the options object along with a time keeper as the model, the settings panel (which I didn't touch except to change some file paths) as the controller, and the badge text, notifications, and audio as three separate views. The views only interact with the state object by registering callbacks for a defined set of events, keeping everything from getting too intertwined.

Also, in this process, I moved the scripts and images into their own folders since I added a couple more js files and the directory started to get cluttered.

I'm sorry if this is too overreaching for one request or if I've done anything improper - this is my first pull  request. I hope at least some of it is usable or helpful, and I'll be continuing to use the addon either way! :)

### CODE REVIEW

File: scripts/options.js [hunk 1]
```diff
@@ -71,6 +71,7 @@ document.getElementById('no-kk').onclick = saveOptions;
 document.getElementById('enable-kk').onclick = saveOptions;
 document.getElementById('always-kk').onclick = saveOptions;
 document.getElementById('enable-notifications').onclick = saveOptions;
+document.getElementById('enable-town-tune').onclick = saveOptions;
```
Comment (kowsen) [inline scripts/options.js:74]: This line is necessary to trigger new option retrieval when the town tune is enabled or disabled. I couldn't find any reason why it wouldn't be here, so I assume it was just an accidental omission.

**Comment by BordenJardine** [comment]
I looked it over (but haven't tested it).
I'm a fan of this commit!

**Comment by GammaGames** [comment]
This would have made the stuff I did on my own fork a lot easier! I've tested it a bit and it seems to be all working, I like it!

edit: Changing notification settings and volume seem to not work

**Comment by kowsen** [comment]
Both are working correctly for me. Are you getting any errors in the console when you change either of those settings?

Edit: Some additional questions - When you change the volume, is it not updating even when you let go of the slider? Are notifications stuck on or off for you?

**Comment by GammaGames** [comment]
The notifications were stuck on and the volume was always half way. I re downloaded it, re added, and restarted Chrome. That appears to have fixed it all!

**Comment by kowsen** [comment]
Phew! That's a relief.

**Comment by kowsen** [comment]
@JdotCarver Is there anything you want me to change or fix with my branch to potentially get this pull request accepted, or should I stop my work on it?

**Comment by JdotCarver** [comment]
I had no idea this was an actual pull request sorry. I usually give rights and the devs merge themselves, I'll merge now.

**Pull request merged**